API/Controllers/BooksController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using VL.Shared.Interfaces;
+using VL.Shared.Model;
+
+namespace API.Controllers
+{
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
+    [Route("api/vl")]
+    [ApiController]
+    public class BooksController : ControllerBase
+    {
+        private readonly IBookService _bookService;
+
+        public BooksController(IBookService bookService)
+        {
+            _bookService = bookService;
+        }
+
+        /// <summary>
+        /// Get all books
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet, Authorize("IsAllUsers")]
+        [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> GetBooks()
+        {
+            var books = await _bookService.GetBooksAsync();
+            if (books == null) return BadRequest();
+            return Ok(books);
+        }
+
+        /// <summary>
+        /// Get a specific book by id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet("{id}"), Authorize("IsAllUsers")]
+        public async Task<ActionResult<Book>> GetBook(int id)
+        {
+            var book = await _bookService.GetBookAsync(id);
+            if (book == null) return BadRequest();
```

Comment:
No change required but C# 10 syntax also lets you do

if(**book is null)** {
return BadRequest();
}

Reply:
Sorry I only code in C#16, let me know when you catch up